### PR TITLE
Min/max values for all types

### PR DESCRIPTION
**Pull request by ptomato**
Closes: #24

### CODE REVIEW

**Comment by ptomato** [comment]
This includes a commit with a proposed resolution for #396, so we should make sure that is the consensus before merging this.

File: polyfill/lib/ecmascript.mjs [hunk 1]
```diff
@@ -743,14 +744,61 @@ export const ES = ObjectAssign(ObjectAssign({}, ES2019), {
...
+    if (year === YEAR_MIN) {
+      try {
+        ES.GetEpochFromParts(year, month, day + 1, hour, minute, second, millisecond, microsecond, nanosecond - 1);
+      } catch (e) {
+        if (!(e instanceof RangeError)) throw e;
+        month = 4;
+        day = 19;
+        hour = minute = second = millisecond = microsecond = 0;
+        nanosecond = 1;
+      }
```
Comment (Ms2ger) [inline polyfill/lib/ecmascript.mjs]: I feel like this implementation relying on the exception type is somewhat brittle; could you review for alternatives?

Comment (ptomato) [reply polyfill/lib/ecmascript.mjs]: Err, of course -1 is not a good out of range value :face_with_head_bandage:
`null` maybe?

**Comment by ptomato** [comment]
It looked like there was general agreement in #24 and #198 about the limits, though since this depends on the fix from #396 I would nonetheless not want to merge this yet. I'll merge the misc. bugfixes to master, but keep the limit commits in this pull request until #396 is resolved.

**Comment by ptomato** [comment]
Rebased on #396.

**Pull request merged**